scipy/sparse/linalg/eigen/_svds.py

```diff
@@ -163,45 +235,28 @@ def svds(A, k=6, ncv=None, tol=0, which='LM', v0=None,
     True
 
     """
-    if which == 'LM':
-        largest = True
-    elif which == 'SM':
-        largest = False
-    else:
-        raise ValueError("which must be either 'LM' or 'SM'.")
 
-    if not (isinstance(A, LinearOperator) or isspmatrix(A)
-            or is_pydata_spmatrix(A)):
-        A = np.asarray(A)
+    args = _iv(A, k, ncv, tol, which, v0, maxiter, return_singular_vectors,
+               solver)
+    A, k, ncv, tol, which, v0, maxiter, return_singular_vectors, solver = args
 
+    largest = (which == 'LM')
     n, m = A.shape
 
-    if k <= 0 or k >= min(n, m):
-        raise ValueError("k must be between 1 and min(A.shape), k=%d" % k)
-
-    if isinstance(A, LinearOperator):
```

Comment:
I just converted everything to a `LinearOperator` in the input validation so we can get rid of this branching.